docs/src/accessibility-testing-java.md

```diff
@@ -0,0 +1,238 @@
+---
+id: accessibility-testing
+title: "Accessibility testing"
+---
+
+Playwright can be used to test your application for many types of accessibility issues.
+
+A few examples of problems this can catch include:
+- Text that would be hard to read for users with vision impairments due to poor color contrast with the background behind it
+- UI controls and form elements without labels that a screen reader could identify
+- Interactive elements with duplicate IDs which can confuse assistive technologies
+
+The following examples rely on the [`com.deque.html.axe-core/playwright`](https://mvnrepository.com/artifact/com.deque.html.axe-core/playwright) Maven package which adds support for running the [axe accessibility testing engine](https://www.deque.com/axe/) as part of your Playwright tests.
+
+<!-- TOC -->
+
+## Disclaimer
+
+Automated accessibility tests can detect some common accessibility problems such as missing or invalid properties. But many accessibility problems can only be discovered through manual testing. We recommend using a combination of automated testing, manual accessibility assessments, and inclusive user testing.
+
+For manual assessments, we recommend [Accessibility Insights for Web](https://accessibilityinsights.io/docs/web/overview/?referrer=playwright-accessibility-testing-java), a free and open source dev tool that walks you through assessing a website for [WCAG 2.1 AA](https://www.w3.org/WAI/WCAG21/quickref/?currentsidebar=%23col_customize&levels=aaa) coverage.
+
+## Example accessibility tests
+
+Accessibility tests work just like any other Playwright test. You can either create separate test cases for them, or integrate accessibility scans and assertions into your existing test cases.
+
+The following examples demonstrate a few basic accessibility testing scenarios.
+
+### Example 1: Scanning an entire page
+
+This example demonstrates how to test an entire page for automatically detectable accessibility violations. The test:
+1. Imports the `com.deque.html.axe-core/playwright` package
+1. Uses normal JUnit 5 `@Test` syntax to define a test case
+1. Uses normal Playwright syntax to open a browser and navigate to the page under test
+1. Invokes `AxeBuilder.analyze()` to run the accessibility scan against the page
+1. Uses normal JUnit 5 test assertions to verify that there are no violations in the returned scan results
+
+```java
+import com.deque.html.axecore.playwright.*; // 1
+import com.deque.html.axecore.utilities.axeresults.*;
+
+import org.junit.jupiter.api.*;
+import com.microsoft.playwright.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class HomepageTests {
+  @Test // 2
+  void shouldNotHaveAutomaticallyDetectableAccessibilityIssues() throws Exception {
+    Playwright playwright = Playwright.create();
```

Comment:
nit: wrap it into `try (Playwright playwright = Playwright.create()) {` to ensure that playwright is closed after running the test.